crt/exit.c

```diff
@@ -52,6 +58,15 @@ void exit(int code)
                     _Exit(code);
                 }
             }
+
+            if (arg.is_child_fork)
+            {
+                // Decrement MUSL's number of threads. Typically this is done by
+                // MUSL's pthread_exit which is not called for child process's
+                // main thread.
+                __atomic_sub_fetch(
```

Comment:
crt protects this with locks. Therefore this is not thread safe. You will need to patch musl to make the variable atomic.

Reply:
Good point. We need to use the same lock that MUSL uses or patch musl to make this atomic.

Reply:
Fixed.

Reply:
Used locks to be consistent with MUSL. I considered using atomics, but felt it was simpler to just follow what MUSL is doing.